components/menu/Menu.js

```diff
@@ -17,162 +17,191 @@ const POSITION = {
 };
 
 class Menu extends PureComponent {
-  state = {
-    active: this.props.active,
-  };
+  state = {};
```

Comment:
Not sure if
```js
state = {};
```
has any functionality, isn't this default?

Then again, I think we're sure the initial state will always be `active: false`, right? Might as well be that then?

Reply:
Correct and correct!

Reply:
`active` in the state, takes the value of the `active` prop, so the menu can be opened from its parent container, `getDerivedStateFromProps` takes care of this. `getDerivedStateFromProps` is also called before the initial mount, so it makes sure that the value of `active` in the state before the mount, is set correctly (aka set to the value of the (default)prop `active`).

Removing the `state = {};` line, throws following error: `Cannot read property 'x' of null`, when calling the `state`. So I think it still needs to be declared.

Reply:
@nickwaelkens @InstaK what do you guys think?